src/components/button-swap/button-swap.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Helper to get the button from the supplied wrapper element
+ */
+const getButton = (wrapper: HTMLElement) =>
+  wrapper.querySelector('.js-c-button-swap__button') as HTMLButtonElement;
+
+/**
+ * Helper to get the status message from the supplied wrapper element
+ */
+const getStatusMessage = (wrapper: HTMLElement) =>
+  wrapper.querySelector('.js-c-button-swap__message') as HTMLElement;
+
+/**
+ * Button swap
+ *
+ * Swaps two buttons by toggling the `hidden` attribute on the wrapper for each
+ * button + visually hidden message group; only one button + visually hidden
+ * message is displayed at a time.
+ */
+export const initButtonSwap = (
+  buttonSwapEl: HTMLElement,
+  {
+    firstBtnCallback,
+    secondBtnCallback,
+  }: {
+    firstBtnCallback?: (event: Event) => void;
+    secondBtnCallback?: (event: Event) => void;
+  } = {}
+) => {
+  // The group wrappers
+  const firstBtnWrapper = buttonSwapEl.querySelector(
+    '.js-c-button-swap__first-btn-wrapper'
+  ) as HTMLElement;
+  const secondBtnWrapper = buttonSwapEl.querySelector(
+    '.js-c-button-swap__second-btn-wrapper'
+  ) as HTMLElement;
+
+  // The buttons
+  const firstBtn = getButton(firstBtnWrapper);
+  const secondBtn = getButton(secondBtnWrapper);
+
+  /**
+   * Performs all firstBtn click actions
+   */
+  const onFirstBtnClick = (event: Event) => {
+    firstBtnWrapper.hidden = true;
+    secondBtnWrapper.hidden = false;
+
+    const statusMsg = getStatusMessage(secondBtnWrapper);
+    statusMsg.setAttribute('role', 'alert');
+    statusMsg.focus();
+
+    if (firstBtnCallback) {
+      firstBtnCallback(event);
+    }
+  };
+
+  /**
+   * Performs all secondBtn click actions
+   */
+  const onSecondBtnClick = (event: Event) => {
+    secondBtnWrapper.hidden = true;
+    firstBtnWrapper.hidden = false;
+
+    const statusMsg = getStatusMessage(firstBtnWrapper);
+    statusMsg.setAttribute('role', 'alert');
+    statusMsg.focus();
+
+    if (secondBtnCallback) {
+      secondBtnCallback(event);
+    }
+  };
```

Comment:
I wonder if both buttons could call the same method and pass in some vars? Something like this?

```js
  const onButtonClick = ({event, hiddenWrapper, visibleWrapper, callback}) => {
    hiddenWrapper.hidden = true;
    visibleWrapper.hidden = false;

    const statusMsg = getStatusMessage(visibleWrapper);
    statusMsg.setAttribute('role', 'alert');
    statusMsg.focus();

    if (callback) {
      callback(event);
    }
  };
  
firstBtn.addEventListener('click', (event) => {
  onButtonClick({
    event,
    hiddenWrapper: firstBtnWrapper,
    visibleWrapper: secondBtnWrapper,
    callback: firstBtnCallback
  });
});
```

That might dry things up a bit, and reduce the amount of duplicated code we're shipping

Reply:
(Though I suppose it would make the `destroy` logic a bit more complicated.)

Reply:
@Paul-Hebert Something like the following?

https://github.com/cloudfour/cloudfour.com-patterns/pull/1516/commits/714e30c29dbc0f5fe5fef9102623790ecb464a30

Reply:
Exactly!